Source/Dafny/Resolver.cs

```diff
@@ -13546,10 +13546,16 @@ private class RBranchExpr : RBranch {
           callee.IsStatic ? null : s.Receiver);
         // type check the out-parameter arguments (in-parameters were type checked as part of ResolveActualParameters)
         for (int i = 0; i < callee.Outs.Count && i < s.Lhs.Count; i++) {
-          var it = callee.Outs[i].Type;
+          var outFormal = callee.Outs[i];
+          var it = outFormal.Type;
           Type st = SubstType(it, typeMap);
           var lhs = s.Lhs[i];
-          AddAssignableConstraint(s.Tok, lhs.Type, st, "incorrect type of method out-parameter" + (callee.Outs.Count == 1 ? "" : " " + i) + " (expected {1}, got {0})");
+          AddAssignableConstraint(
+            s.Tok, lhs.Type, st, "incorrect type of method out-parameter" +
+                                 (callee.Outs.Count == 1 ? "" : " " + i) +
```

Comment:
It seems redundant to specify both the index and the name of the parameter.

Reply:
It seems indeed, but in Dafny you can omit the name of datatype arguments, so the index is still needed, and the name helps a lot (especially when checking the meaning).
This PR takes an hybrid approach: As before, do not show the index if there is only one argument; do not show the name if it has been derived implicitly,
Looking at the error messages, I don't personally feel bothered by seeing the argument index (where to look at) and the name (the semantics of this argument) at the same time.